docs/cli/index.md

```diff
@@ -0,0 +1,108 @@
+opinionated tooling, delivered primarily through the
+[`@backstage/cli`](https://www.npmjs.com/package/@backstage/cli) package.
+
+The `@backstage/cli` package provides a single executable script,
+`backstage-cli`, which you can run directly with `yarn` or within a script in
+`package.json`. If you have a Backstage app set up, you can try out the
+following command to print the top-level help page of the CLI:
+
+```text
+yarn backstage-cli --help
+```
+
+If you are familiar with [`create-react-app`](https://create-react-app.dev/) you
+may recognize the pattern of bundling tooling up as a CLI, as it uses a package
+called [`react-scripts`](https://www.npmjs.com/package/react-scripts) to bring
+most of the functionality into the created project. The Backstage equivalent of
+`create-react-app` is
+[`@backstage/create-app`](https://www.npmjs.com/package/@backstage/create-app),
+and the equivalent of `react-scripts` is `@backstage/cli`. There are however a
+couple of key differences between the two. Most notably, Backstage apps are
+monorepos and the CLI is tailored for that environment. It provides tooling both
+for bundling and developing full end-user apps, but also for developing,
+building and publishing individual packages within the monorepo, as well as
+tooling that is more unique to Backstage, such as commands for working with
+static configuration.
+
+## Opinionated Tooling
+
+The Backstage CLI is highly opinionated in what tools are used and how they are
+configured. It is tailored for development in large TypeScript monorepos with
+hundreds of separate packages, but with the ability to have edits anywhere in
+the codebase reflected within a few seconds. The build output is also optimized
+for this setup, and aims to provide an excellent user experience with fast page
+load times in modern browsers, rather than a wide range of support.
+
+While the Backstage tooling is opinionated in how to develop and build packages,
+it is also possible to use your own tooling either partially or fully. For
+example, the CLI provides a command for building a plugin package for
+publishing, but the output is a quite standard combination of transpiled
+JavaScript and TypeScript type declarations. The usage of the command from the
+CLI can therefore easily be replaced with other tools if necessary.
+
+Just like `react-scripts`, the Backstage CLI does not provide many hooks for
+overriding or customizing the build process. This is to allow for evolution of
+the CLI without having to take a wide API surface into account. Allowing us to
+quickly iterate and improve the tooling, as well more easily keep dependencies
+up to date.
+
+## Opinions & Goals
+
+In no particular order, this is a list of opinions and goals that guide the
+design and development of the Backstage CLI:
+
+- All you need for development is `yarn start`, there should be no need to
+  manually build packages or run other separate tasks.
```

Comment:
well there's `install` and `tsc`, but yeah :)

Reply:
Install is tricky to get around :p, but also only needs to be run once. `tsc` isn't needed